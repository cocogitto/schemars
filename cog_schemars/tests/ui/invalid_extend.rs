use cog_schemars::JsonSchema;

#[derive(JsonSchema)]
#[cog_schemars(extend(x))]
#[cog_schemars(extend("x"))]
#[cog_schemars(extend("x" = ))]
#[cog_schemars(extend("y" = "ok!", "y" = "duplicated!"), extend("y" = "duplicated!"))]
#[cog_schemars(extend("y" = "duplicated!"))]
pub struct Struct;

fn main() {}
