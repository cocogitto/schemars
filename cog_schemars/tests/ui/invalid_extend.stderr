error: expected string literal
 --> tests/ui/invalid_extend.rs:4:23
  |
4 | #[cog_schemars(extend(x))]
  |                       ^

error: expected `=`
 --> tests/ui/invalid_extend.rs:5:26
  |
5 | #[cog_schemars(extend("x"))]
  |                          ^

error: Expected extension value
 --> tests/ui/invalid_extend.rs:6:29
  |
6 | #[cog_schemars(extend("x" = ))]
  |                             ^

error: Duplicate extension key 'y'
 --> tests/ui/invalid_extend.rs:7:36
  |
7 | #[cog_schemars(extend("y" = "ok!", "y" = "duplicated!"), extend("y" = "duplicated!"))]
  |                                    ^^^

error: Duplicate extension key 'y'
 --> tests/ui/invalid_extend.rs:7:65
  |
7 | #[cog_schemars(extend("y" = "ok!", "y" = "duplicated!"), extend("y" = "duplicated!"))]
  |                                                                 ^^^

error: Duplicate extension key 'y'
 --> tests/ui/invalid_extend.rs:8:23
  |
8 | #[cog_schemars(extend("y" = "duplicated!"))]
  |                       ^^^
